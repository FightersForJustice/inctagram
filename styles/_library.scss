//vars
//mainColor

$bgColor: #0d0d0d;
$text: #fff;

@mixin btn {
    border-radius:2px;
    background: #397DF6;
    color: #FFF;
    font-size: 16px;
    font-family: Inter;
    font-style: normal;
    font-weight: 600;
    line-height: 24px;
    cursor: pointer;
  }