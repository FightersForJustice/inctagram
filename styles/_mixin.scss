

@mixin Primary-btn {
    border-radius:2px;
    background: #397DF6;
    color: #FFF;
    font-size: 16px;
    font-family: Inter;
    font-style: normal;
    font-weight: 600;
    line-height: 24px;
    cursor: pointer;
  }
@mixin Secondary-btn {
    cursor: pointer;
  }
@mixin Outline-btn {
  cursor: pointer;
}
@mixin Large {
  color:  #000;
  font-size: 26px;
  font-style: normal;
  font-weight: 600;
  line-height: 36px;
}
@mixin H1 {
  color:  #000;
  font-size: 20px;
  font-style: normal;
  font-weight: 700;
  line-height: 36px;
}
@mixin H2 {
  color:  #000;
  font-size: 18px;
  font-style: normal;
  font-weight: 700;
  line-height: 24px;
}
@mixin H3 {
  color: #000;
  font-family: Inter;
  font-size: 16px;
  font-style: normal;
  font-weight: 600;
  line-height: 24px;  
}
@mixin regular_text_16 {
  color: $dark-900;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
}
@mixin Bold_text_16 {
  color: $dark-900;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  line-height: 24px;
}
@mixin regular_text_14 {
  color: $dark-900;
  font-size: 14px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
}
@mixin Medium_text_14 {
  color: $dark-900;
  font-size: 14px;
  font-style: normal;
  font-weight: 500;
  line-height: 24px;
}
@mixin bold_text_14 {
  color: $dark-900;
  font-size: 14px;
  font-style: normal;
  font-weight: 700;
  line-height: 24px;
}
@mixin small_text {
  color: $dark-900;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 16px;
}
@mixin Semi-bold_small-text {
  color: $dark-900;
  font-size: 12px;
  font-style: normal;
  font-weight: 600;
  line-height: 16px;
}
@mixin regular_link {
  color: $primary-500;
  font-size: 14px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  text-decoration-line: underline;
}
@mixin small_link {
  color: $primary-500;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 16px;
  text-decoration-line: underline;
}