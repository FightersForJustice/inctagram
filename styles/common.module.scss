@use 'library';

.container {
  background-color: library.$dark-700;
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.title {
  color: library.$light-100;
  text-align: center;
  text-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
  font-size: 20px;
  font-family: Inter;
  font-style: normal;
  font-weight: 700;
  line-height: 36px;
}

.text {
  color: library.$light-100;
  text-align: center;
  font-size: 16px;
  font-family: Inter;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
}
