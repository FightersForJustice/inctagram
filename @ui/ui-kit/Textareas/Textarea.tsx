import React, { ChangeEvent, FocusEvent } from 'react'
import classNames from 'classnames'
import styles from './Textarea.module.scss'
import { TEXTAEREA_COLORS } from './constants'

export type TextAreaProps = {
  hasError?: boolean
  disabled?: boolean
  hovered?: boolean
  errorMessage?: string
  placeholder?: string
  color?: (typeof TEXTAEREA_COLORS)[keyof typeof TEXTAEREA_COLORS]
} & React.TextareaHTMLAttributes<HTMLTextAreaElement> // Inherit all textarea HTML attributes

export const TextArea: React.FC<TextAreaProps> = ({
  defaultValue,
  hasError = false,
  disabled = false,
  hovered = false,
  errorMessage = 'Error text',
  placeholder = 'Textarea',
  color = TEXTAEREA_COLORS.DEFAULT,
  ...restProps // Spread the remaining HTML attributes
}) => {
  const textareaClasses = classNames(styles.textarea, {
    [styles.Default]: color === TEXTAEREA_COLORS.DEFAULT,
    [styles.Active]: color === TEXTAEREA_COLORS.ACTIVE,
    [styles.Error]: color === TEXTAEREA_COLORS.ERROR,
  })

  return (
    <div>
      <p className={styles.title}>Textarea</p>
      <textarea
        className={textareaClasses}
        defaultValue={defaultValue}
        disabled={disabled}
        placeholder={placeholder}
        {...restProps} // Spread the remaining HTML attributes
      />
      {hasError && <p className={styles.error_message}>{errorMessage}</p>}
    </div>
  )
}
