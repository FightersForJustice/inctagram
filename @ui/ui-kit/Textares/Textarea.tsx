import React from 'react'
import { useController, Control } from 'react-hook-form'
import classNames from 'classnames'
import styles from './Textarea.module.css'

interface TextAreaProps {
  name: string
  control: Control
  defaultValue?: string
  rules?: Object
  disabled?: boolean
  isDirty?: boolean
  isFocused?: boolean
  isHovered?: boolean
}

const TextArea: React.FC<TextAreaProps> = ({ name, control, defaultValue, rules, ...rest }) => {
  const {
    field: { ref, value, onChange, onBlur },
    fieldState: { error },
  } = useController({
    name,
    control,
    defaultValue,
    rules,
  })

  const hasError = !!error

  const className = classNames('textarea', {
    disabled: rest.disabled,
    error: hasError,
    active: rest.isDirty,
    focus: rest.isFocused,
    hover: rest.isHovered,
  })

  return (
    <div>
      <textarea className={className} ref={ref} value={value} onChange={onChange} onBlur={onBlur} {...rest} />
      {hasError && <span className="error-message">{error.message}</span>}
    </div>
  )
}

export default TextArea
