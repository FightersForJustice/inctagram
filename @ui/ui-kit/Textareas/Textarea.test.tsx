import React from 'react'
import { render, fireEvent, cleanup, screen, waitFor } from '@testing-library/react'
import { TextArea } from './Textarea'
import styles from './Textarea.module.scss'

const value = 'type sth here...'

describe('Textarea', () => {
  const setup = (props = {}) => {
    const defaultProps = {
      onChange: jest.fn(),
      onFocus: jest.fn(),
      disabled: false,
      ...props,
    }

    render(<TextArea {...defaultProps} />)

    return {
      textarea: screen.getByRole('textbox'),
      ...defaultProps,
    }
  }

  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    const { textarea } = setup()

    expect(textarea).toBeInTheDocument()
    expect(textarea).toMatchSnapshot()
  })

  it('changes value and calls onChange', () => {
    const { textarea, onChange } = setup()
    fireEvent.change(textarea, { target: { value } })

    expect(textarea).toHaveValue(value)
    expect(onChange).toHaveBeenCalledTimes(1)
  })

  //   it('does not respond to events when disabled', () => {
  //     const { textarea, onChange, onFocus } = setup({ disabled: true })

  //     fireEvent.change(textarea, { target: { value } })
  //     fireEvent.focus(textarea)

  //     expect(onChange).not.toHaveBeenCalled()
  //     expect(onFocus).not.toHaveBeenCalled()
  //   })

  it('is disabled when disabled prop is true', () => {
    const { textarea } = setup({ disabled: true })

    expect(textarea).toBeDisabled()
  })
  it('is not disabled when disabled prop is missing', () => {
    const { textarea } = setup()

    expect(textarea).not.toBeDisabled()
  })
  it('is not disabled when disabled prop is false', () => {
    const { textarea } = setup({ disabled: false })

    expect(textarea).not.toBeDisabled()
  })

  it('has default styles when color prop is not provided', () => {
    const { textarea } = setup()

    expect(textarea).toHaveClass(styles.Default)
  })
  it('has Active styles when Active color prop is provided', () => {
    const { textarea } = setup({ color: 'Active' })

    expect(textarea).toHaveClass(styles.Active)
    expect(textarea).toMatchSnapshot()
  })
  it('has Error styles when Error color prop is provided', () => {
    const { textarea } = setup({ color: 'Error' })

    expect(textarea).toHaveClass(styles.Error)
    expect(textarea).toMatchSnapshot()
  })
})
