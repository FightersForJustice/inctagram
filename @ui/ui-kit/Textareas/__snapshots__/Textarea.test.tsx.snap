// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea has Active styles 1`] = `
<textarea
  class="textarea Active"
  placeholder="Textarea"
/>
`;

exports[`Textarea has Active styles when Active color prop is not provided 1`] = `
<textarea
  class="textarea Active"
  placeholder="Textarea"
/>
`;

exports[`Textarea has Active styles when Active color prop is provided 1`] = `
<textarea
  class="textarea Active"
  placeholder="Textarea"
/>
`;

exports[`Textarea has Error styles 1`] = `
<textarea
  class="textarea Error"
  placeholder="Textarea"
/>
`;

exports[`Textarea has Error styles when Error color prop is not provided 1`] = `
<textarea
  class="textarea Error"
  placeholder="Textarea"
/>
`;

exports[`Textarea has Error styles when Error color prop is provided 1`] = `
<textarea
  class="textarea Error"
  placeholder="Textarea"
/>
`;

exports[`Textarea renders correctly 1`] = `
<textarea
  class="textarea Default"
  placeholder="Textarea"
/>
`;
